File: Backend/app/schemas/calendar.py
from app.core.serialization import CamelModel
from app.core.timeutils import UtcDatetime
from app.schemas.common import CalendarStatus


class CalendarConnection(CamelModel):
    connected: bool
    provider: str | None
    account_email: str | None
    scopes: list[str]
    last_sync_at: UtcDatetime | None
    status: CalendarStatus | None


class ConnectCalendarResponse(CamelModel):
    authorization_url: str

File: Frontend/src/types/contracts/calendar.ts
import { z } from "zod";

export const calendarConnectionSchema = z.object({
  connected: z.boolean(),
  provider: z.literal("google").nullable(),
  accountEmail: z.string().nullable(),
  scopes: z.array(z.string()),
  lastSyncAt: z.string().nullable(),
  status: z.enum(["healthy", "expired", "error"]).nullable(),
});
